lib/decompress/zstd_decompress.c

```diff
@@ -306,6 +306,7 @@ unsigned long long ZSTD_getDecompressedSize(const void* src, size_t srcSize)
 static size_t ZSTD_decodeFrameHeader(ZSTD_DCtx* dctx, const void* src, size_t srcSize)
 {
     size_t const result = ZSTD_getFrameParams(&(dctx->fParams), src, srcSize);
+    if (result > 0) return ERROR(srcSize_wrong);  /* Don't access fParams if call to ZSTD_getFrameParams is not successful */
```

Comment:
It would be better to use `CHECK_F(ZSTD_getFrameParams(&(dctx->fParams), src, srcSize));`